test/unit/ecs/systems/RenderSystem.test.ts

```diff
@@ -11,4 +11,23 @@ describe('systems/RenderSystem', () => {
             expect(renderSystem.components).toContain(myTestShape);
         })
     })
+
+    describe('.update()', () => {
+        const renderSystem = new RenderSystem(new Renderer(new CanvasDevice()));
+
+        it('Should draw renderable object', () => {
+            const drawSpy = jest.spyOn(ShapeComponent.prototype, 'draw');
+
+            renderSystem.registerComponent(new ShapeComponent());
+            renderSystem.registerComponent(new ShapeComponent());
+
+            renderSystem.update();
+
+            expect(drawSpy).toHaveBeenCalledTimes(2);
+        });
+
+        it.todo('Should render objects based on their depthIndex in reverse order (0 rendered last)')
+
+        it.todo('Should render object based on their transparency (transparent last)')
```

Comment:
add another test case like it('Should render transparency objects as last but ordered by depth index)